servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionObserver.java

```diff
@@ -79,4 +95,135 @@
          */
         void handshakeComplete(SSLSession sslSession);
     }
+
+    /**
+     * An observer interface that provides visibility into read and write events of a non-multiplexed connection.
+     */
+    interface NonMultiplexedObserver {
```

Comment:
I would prefer to avoid the word `Factory` and keep just an `Observer` suffix everywhere for consistency. Observer provides callbacks for events, we don't need to be too specific for events that return another observer.